presto-hive/src/test/java/com/facebook/presto/hive/TestShowStats.java

```diff
@@ -144,12 +152,13 @@ public void testShowStatsWithSelectDistinctFails()
     @Test
     public void testShowStatsWithSelectFunctionCallFails()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT sin(orderkey) FROM orders)", ".*Only \\* and column references are supported by SHOW STATS SELECT clause");
+        assertQueryFails("SHOW STATS FOR (SELECT sin(orderkey) FROM orders)", ".*Only SELECT \\* is supported in SHOW STATS SELECT clause");
+        assertQueryFails("SHOW STATS FOR (SELECT count(*) FROM orders)", ".*Only SELECT \\* is supported in SHOW STATS SELECT clause");
     }
 
     @Test
     public void testShowStatsWithWhereFunctionCallFails()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT orderkey FROM orders WHERE sin(orderkey) > 0)", ".*Only literals, column references, comparators, is \\(not\\) null and logical operators are allowed in WHERE of SHOW STATS SELECT clause");
+        assertQueryFails("SHOW STATS FOR (SELECT * FROM orders WHERE sin(orderkey) > 0)", ".*Only literals, column references, comparators, is \\(not\\) null and logical operators are allowed in WHERE of SHOW STATS SELECT clause");
```

Comment:
do you think we are going to support `SELECT <anything other than *>` in foreseeable future? If not why not change the syntax to: 

```
SHOW STATS FOR orders WHERE sin(orderkey) > 0
```

That is much straightforward and easier. `SELECT` suggests that something may work, while it does not (besides `*`).

Maybe it is another story for followup. What do you think?

CC: @mbasmanova @findepi

Reply:
@kokosing I'd like to add support for arbitrary queries in SHOW STATS at some point. At a minimum, it would be helpful to support a list of columns. Otherwise, the output for wide tables is hard to consume.

Reply:
Sounds good to me. So we need to make sure that error messages we are raising tell that this is not yet supported.